Dockerfile

```diff
@@ -0,0 +1,67 @@
+# This is a multi-stage build which requires Docker 17.05 or higher
+FROM python:3.7-alpine as molecule-builder
```

Comment:
Could you please add a couple of lines saying why a multi-stage build is needed, will be useful to future maintainers.

Reply:
It's just to have a "fat" build env in a different container and copy the built artifacts (wheels) into the target one without having to keep the build deps there. I was trying hard to make the resulting image smaller.
I wish I could do the same with Ruby deps but I don't really know it and I have no idea whether it's even possible.

Reply:
Is it not just another build stage? You could try https://hub.docker.com/_/ruby?tab=tags? It looks like https://github.com/docker-library/ruby/blob/eae22dc2dfe30b48f1633912396110539e1b8d49/2.5/alpine3.9/Dockerfile matches the Alpine version in the Python images we're using https://github.com/docker-library/python/blob/9bb5d46dbcf16250a2d292ddc1e0b7a792aa275f/3.7/alpine3.9/Dockerfile.

Reply:
It's a good idea but I don't know how to do the same for ruby